docs/data/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,214 @@
+
+It’s important to anticipate errors your users may encounter as you develop on Stellar. In many tutorials throughout our developer documentation, we leave out error handling code to focus on the example. In this section, we will do the opposite and talk specifically about the errors. By the end of this section, you should be able to categorize errors and understand the best way to handle them in your application.
+
+Many actions interact with the Stellar network through the Horizon API, and these possible actions fall into two main categories:
+
+1. Queries (any `GET` request, like to `/accounts`)
+2. Transaction submissions (a `POST /transactions`, `POST /transactions_async`).
+
+There are many possible error codes when executing these actions, and you can typically handle these error codes using the following strategies:
+
+- Request adjustments: adjusting the request to resolve structural errors with queries or transaction submissions. Suppose you’ve included a bad parameter, malformed your XDR, or otherwise didn’t follow the endpoint’s specification. In these cases, resolve the error by referencing the details or result codes of the error response.
+- Polling and retrying: this is the recommended way to work around latency or congestion issues encountered along the pipeline between your computer and the Stellar network, which can sometimes happen due to the nature of distributed systems.
+
+## Error Handling for Queries
+
+Many `GET` requests have specific parameter requirements, and while the SDKs can help enforce them, you can still pass invalid arguments (for example, an asset string that isn’t SEP-11 compatible) that error out every time. In this scenario, there’s nothing you can do aside from following the API specification. The `extras` field of the error response will often clue you in on where to look and what to look for.
+
+<CodeExample>
+
+```bash
+curl -s https://horizon-testnet.stellar.org/claimable_balances/0000 | jq '.extras'
+{
+    "invalid_field": "id",
+    "reason": "Invalid claimable balance ID"
+}
+```
+
+</CodeExample>
+
+Note that the SDKs make it a point to distinguish an invalid request (as above) versus a missing resource (a `404 Not Found`) (for example, the generic `NetworkError` versus a `NotFoundError` in the JavaScript SDK), where the latter might not be considered an error depending on your situation.
+
+## Error Handling for Transaction Submissions
+
+Horizon currently supports two types of transaction submission endpoints:
+
+1. `/transactions_async`: Horizon submits a transaction to Stellar-Core in an asynchronous manner and passes the relevant Stellar-Core response back immediately to the client. It is then the client's responsibility to poll for the status of that transaction.
+2. `/transactions`: Horizon submits a transaction to Stellar-Core and then waits for it to be ingested in the database. It will either timeout while waiting or return a transaction response on success.
```

Comment:
A couple things:

- In the async case we mention that its the client's responsibility to poll after getting a response from the submission endpoint. Can we also mention this for the sync case in the event of a timeout?
- The sync case describes the behavior of the endpoint to return timeouts or success responses, but its missing a mention that a transaction failure response could be returned too.

```suggestion
1. `/transactions_async`: Horizon submits a transaction to Stellar-Core in an asynchronous manner and passes the relevant Stellar-Core response back immediately to the client. It is then the client's responsibility to poll for the status of that transaction.
2. `/transactions`: Horizon submits a transaction to Stellar-Core and then waits for it to be ingested into its database. It will either return a success (200) or failure (400) response, or return a timeout response (504), after which it is the client's responsibility to poll for the status of the transaction.
```